src/test/java/edu/cornell/kfs/sys/service/impl/CuUserFavoriteAccountServiceImplTest.java

```diff
@@ -1,105 +1,136 @@
-    private UserFavoriteAccountService userFavoriteAccountService;
-    @Override
-    protected void setUp() throws Exception {
-        // TODO Auto-generated method stub
-        super.setUp();
-        userFavoriteAccountService = SpringContext.getBean(UserFavoriteAccountService.class);
+    @Before
+    public void setUp() throws Exception {
+        userFavoriteAccountService = new TestUserFavoriteAccountServiceImpl();
+        userFavoriteAccountService.setBusinessObjectService(new MockBusinessObjectServiceImpl());
     }
 
-    /*
-     * test get user primary favorite account
-     * 
-     */
+    @Test
     public void testPrimaryFavoriteAccount() {
-        Map<String, String> fieldMap = new HashMap<String, String>();
-        fieldMap.put("primaryInd", "Y");
-        List<FavoriteAccount> favoriteAccounts = (List<FavoriteAccount>)SpringContext.getBean(BusinessObjectService.class).findMatching(FavoriteAccount.class, fieldMap);
+        List<FavoriteAccount> favoriteAccounts = buildFavoriteAccountList();
         FavoriteAccount favoriteAccount = userFavoriteAccountService.getFavoriteAccount(favoriteAccounts.get(0).getUserProcurementProfile().getPrincipalId());
-        assertTrue("Favorite Account should exist", favoriteAccount != null);
-        assertTrue("should be Primary favorite account", favoriteAccount.getPrimaryInd());
+        Assert.assertTrue("Favorite Account should exist", favoriteAccount != null);
+        Assert.assertTrue("should be Primary favorite account", favoriteAccount.getPrimaryInd());
         
         favoriteAccount = userFavoriteAccountService.getFavoriteAccount("2");
-        assertTrue("Favorite Account should not exist", favoriteAccount == null);
+        Assert.assertTrue("Favorite Account should not exist", favoriteAccount == null);
+    }
+
+    private List<FavoriteAccount> buildFavoriteAccountList() {
+        List<FavoriteAccount> favoriteAccounts = new ArrayList<>();
+
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_1.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_2.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_3.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_4.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_5.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_6.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_7.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_8.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_9.createFavoriteAccount());
+        favoriteAccounts.add(FavoriteAccountFixture.FAVORITE_ACCOUNT_10.createFavoriteAccount());
```

Comment:
The original test loaded all the favorites from the db, so I started with a subset to mimic that behavior. Let me review to see if that still makes sense, or if it can be whittled down to just one.